popup_filter_build.js

```diff
@@ -0,0 +1,55 @@
+const { promises: fs } = require('fs');
+
+/**
+ * Path to the filter file
+ * @type {string}
+ */
+const filterPath = process.argv[2];
+
+/**
+ * Function to convert modifiers in a rule
+ * @param {string} rule - The rule to modify
+ * @param {string} modifier - The new modifier
+ * @param {string} [modifierValue] - The value for the modifier
+ * @returns {string} - Modified rule
+ */
+const convertModifier = (rule, modifier, modifierValue) => {
+    // Avoiding comments
+    if (rule.startsWith('!')) {
+        return rule;
+    }
+    // Add the modifier to the rule
+    return `${rule}$${modifier}${modifierValue ? `=${modifierValue}` : ''}`;
+};
+
+/**
+ * Function to get file content as an array of lines
+ * @param {string} path - Path to the file
+ * @returns {Promise<string[]>} - Promise resolving to an array of lines from the file
+ */
+const getFileContent = async (path) => {
```

Comment:
file content is more about a string return type
the function should be called `getFileRules`

Reply:
or current name may remain as is but return type should be changed
```
/**
 * Returns a file content.
 * @param {string} path Path to the file.
 * @returns {Promise<string>} - Promise resolving to the file content.
 */
const getFileContent = async (path) => {
    try {
        const content = await fs.readFile(path, 'utf-8');
        return content;
    } catch (error) {
        throw new Error(`Error during reading the file '${path}' due to: ${error.message}`);
    }
};
```

and after that `convertFilterList()` can be changed:
```
const LINE_BREAK = '\n';
const COMMENT_MARKER = '!';

const convertFilterList = async (path) => {
    try {
        const fileContent = await getFileContent(path);
        const modifiedContent = fileContent
            .split(LINE_BREAK)
            .map((line) => {
                if (line.startsWith(COMMENT_MARKER)) {
                    return line;
                }
                return `${line}$dnsrewrite=ad-block.dns.adguard.com`;
            })
            .join(LINE_BREAK);
        await fs.writeFile(path, modifiedContent);
    } catch (error) {
        throw new Error(`Error during rules conversion due to: ${error.message}`);
    }
};
```